### DISCUSSION TITLE
Feature Request: InnoDB and Foreign Keys

### DISCUSSION THREAD

**Discussion by Zen4All** [zencart/zencart#5820] [Ideas]
Maybe it is time to move towards InnoDB, so we can make use of Foreign keys. This would making coding for changes to tables easier, because i.e. we could the products table to specials and featured, so when deleting a product, all related tables get updated to, without the need of writing the needed (complicated) queries.
I use it in some of my add ons, and it save a lot of time in coding / troubleshooting.

**Comment by chindraba-work** [comment]
Supposing that there isn't a meaningful segment of hosts where the MySQL database is not able to handle InnoDB tables, this would be a _very_ welcome switch. Upgrading existing databases, with who-can-guess collections of plugins, however, could be a big issue. If there are sufficient other changes in the switch from v1 to v2, a complete database overhaul could be in order anyway.

**Comment by zcwilt** [comment]
One other huge benefit of using innodb is to move towards transactional queries.

**Comment by drbyte** [comment]
> Supposing that there isn't a meaningful segment of hosts where the MySQL database is not able to handle InnoDB tables

While I haven't looked for statistics to prove it, I'm hopeful that the hosting climate these days largely supports InnoDB. 
[The default engine is InnoDB as of MySQL 5.5.5](https://dev.mysql.com/doc/refman/5.5/en/storage-engine-setting.html)

**Comment by zcwilt** [comment]
https://github.com/zencart/zencart/pull/5754
